src/sorting/bubble_sort.rs

```diff
@@ -3,21 +3,29 @@
 // in the wrong order.
 // Time complexity is O(N^2)
 // Auxiliary space is O(1)
+use super::traits::MutableSorter;
 
-pub fn bubble_sort<T: Ord>(array: &mut [T]) {
-    for i in 0..array.len() {
-        // Last i elements are already in place.
-        for j in 0..array.len() - 1 - i {
-            if array[j] > array[j + 1] {
-                array.swap(j, j + 1);
+pub struct BubbleSort;
+
+impl<T> MutableSorter<T> for BubbleSort {
+    fn sort(array: &mut [T])
+    where
+        T: Ord,
+    {
+        for i in 0..array.len() {
+            // Last i elements are already in place.
+            for j in 0..array.len() - 1 - i {
+                if array[j] > array[j + 1] {
+                    array.swap(j, j + 1);
+                }
             }
         }
     }
 }
 
 #[cfg(test)]
 mod tests {
-    use super::bubble_sort;
-
-    sorting_tests!(bubble_sort, inplace);
+    use super::super::traits::MutableSorter;
```

Comment:
Ah, I missed it on the previous PR, but you can do this here since you already require it in the parent scope.
```suggestion
    use super::MutableSorter;
```